### ISSUE TITLE
Docs are wrong anywhere we use shorthand Array notation

### ISSUE THREAD

**Issue by rweber-esri** [Esri/hub.js#285]
Generics Array notation (e.g. `IModel[]`) and shorthand Array notation (e.g. `Array<IModel>`) are semantically equal in TypeScript. Apparently this is not the case, though, for our documentation. The more verbose generics syntax compiles to the expected documentation type of `IModel[]`, however shorthand notation compiles to something like `IModel[] | IModel` which is not right. We use [typedoc](https://github.com/TypeStrong/typedoc) to generate our documentation. Looks like that lib is up to `v0.17.8`, we're currently using `v0.14.2`. We should see if this inconsistency can be corrected by updating the `typedoc` lib and/or tweaking it's configuration. If those efforts fall flat, we should replace all instances of shorthand Array notation with generics notation and investigate enforcing via linting rules.

You can see this issue in action by viewing [IModelTemplate docs](https://esri.github.io/hub.js/api/common/IModelTemplate/). See `assets`, `dependencies` & `resources` properties.

**Comment by dbouwman** [comment]
Thanks @rweber-esri! I'm checking if bumping `typedoc` will resolve. If to, I'll add to a PR I've got open

**Comment by dbouwman** [comment]
@rweber-esri I updated `typedoc` but that was not the problem - rather it seems to have been in the actual doc generation code. I've updated that code to not add the `| type` any more.

![image](https://user-images.githubusercontent.com/119129/86606659-86a84c00-bf65-11ea-8ae7-ae1a6be339b5.png)


I'm not the original author of this, so we will have to see if this has other side-effects. Will tag you on the PR
